src/server/string_family.cc

```diff
@@ -379,31 +379,27 @@ OpStatus OpMSet(const OpArgs& op_args, const ShardArgs& args) {
   return result;
 }
 
-// emission_interval_ms assumed to be positive
+// emission_interval_ns assumed to be positive
 // limit is assumed to be positive
 OpResult<array<int64_t, 5>> OpThrottle(const OpArgs& op_args, const string_view key,
-                                       const int64_t limit, const int64_t emission_interval_ms,
+                                       const int64_t limit, const int64_t emission_interval_ns,
                                        const uint64_t quantity) {
   auto& db_slice = op_args.GetDbSlice();
 
-  if (emission_interval_ms > INT64_MAX / limit) {
-    return OpStatus::INVALID_INT;
-  }
-  const int64_t delay_variation_tolerance_ms = emission_interval_ms * limit;  // should be positive
+  // Total size of the bucket
+  const int64_t delay_variation_tolerance_ns = emission_interval_ns * limit;  // should be positive
 
   int64_t remaining = 0;
   int64_t reset_after_ms = -1000;
   int64_t retry_after_ms = -1000;
 
-  if (quantity != 0 && static_cast<uint64_t>(emission_interval_ms) > INT64_MAX / quantity) {
```

Comment:
same q

Reply:
same as https://github.com/dragonflydb/dragonfly/pull/5686#discussion_r2290558285 this can be done before the transaction and fail early if required